src/providers/comments-provider.tsx

```diff
@@ -20,18 +18,6 @@ export const CommentsProvider: React.VoidFunctionComponent<{
 }> = ({ children }) => {
   const [comments, setComments] = React.useState<Comment[]>([]);
   const [newCommentText, setNewCommentText] = React.useState<string>("");
-  const { accidentId } = useAccident();
-
-  React.useEffect(() => {
-    const initFetchComments = async () => {
-      const initialComments = await fetchComments(
-        window.location.origin,
-        accidentId,
-      );
-      setComments(initialComments);
-    };
-    void initFetchComments();
-  }, [accidentId]);
```

Comment:
I believe we can achieve the same result while keeping our fetch function where it was. E.g.:

```ts
  React.useEffect(() => {
    let effectIsStale = false;
    
    const fetchAndSetComments = async () => {
      const comments = await fetchComments(
        window.location.origin,
        accidentId,
      );

      if (!effectIsStale) {
        setComments(comments);
      }
    };
    
    void fetchAndSetComments();

    return () => {
      effectIsState = true;
    }
  }, [accidentId]);
```

If we move fetching to the list of comments, the value of provider diminishes and it no longer does what it ‘says on the tin’.

Reply:
This move for preventing memory leak (https://github.com/dtpstat/website/issues/48)

Reply:
```
if (!effectIsStale) { setComments(comments); }
```

should prevent it

Reply:
You’ve done pretty much the same change, which is correct. I’m just suggesting to keep `useEffect` in the original component. If you still see a warning, let me know and I’ll play with the code locally 👀

Reply:
Got it, moved.